hubblestack/extmods/hubble_mods/win_reg.py

```diff
@@ -0,0 +1,209 @@
+# -*- encoding: utf-8 -*-
+"""
+Module for fetching registry values from windows registry
+
+Audit Example 1:
+---------------
+check_unique_id:
+  description: 'win_reg check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:Microsoft Windows Server 2016*'
+      hubble_version: '>3 AND <7 AND <8'
+      module: win_reg
+      items:
+        - args:
+            name: 'HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention'
+          comparator:
+            type: "dict"
+            match:
+              HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention:
+                 type: "number"
+                 match: "== 0"
+
+Audit Example 2:
+---------------
+check_unique_id:
+  description: 'win_reg check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:Microsoft Windows Server 2016*'
+      hubble_version: '>3 AND <7 AND <8'
+      module: win_reg
+      items:
+        - args:
+            name: 'HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft Services\AdmPwd\PasswordAgeDays'
+          comparator:
+            type: "dict"
+            match:
+              HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft Services\AdmPwd\PasswordAgeDays:
+                 type: "number"
+                 match: "<= 30"
+
+FDG Example:
+------------
+main:
+  description: 'win_reg fdg'
+  module: win_reg
+  args:
+    name: HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPreventions
+Mandatory parameters:
+    name - registry name
+
+Note: Comparison logic is moved to comparators. Module will just invoke the win_reg command.
+Comparator compatible with this module - number, string
+
+Sample Output:
+1. can be dictionary with value being an int like
+    {"HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention": 0}
+2. can be a dictionary with value being a str like
+    {"HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\NetworkProvider\HardenedPaths\\*\NETLOGON":
+                                'RequireMutualAuthentication=1,RequireIntegrity=1'}
+
+Note: In normal execution, this module expects a registry name.
+In case of chaining, it expects a string(registry name) from chaining
+"""
+
+import logging
+import salt.utils
+import salt.utils.platform
+
+import hubblestack.extmods.module_runner.runner_utils as runner_utils
+from hubblestack.utils.hubble_error import HubbleCheckValidationError
+
+
+log = logging.getLogger(__name__)
+
+def __virtual__():
```

Comment:
instead of having this function here, we should validate it in validate_params

Reply:
The intention of the function is not to validate params. It is to figure out if the module should be loaded or not.